File: angular-frontend/src/app/core/services/saved-search.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { SavedSearchRequest, SavedSearchResponse } from '../models/saved-search.model';
import { ApiResponse } from '../models/tender.model';

@Injectable({ providedIn: 'root' })
export class SavedSearchService {
  private baseUrl = environment.apiUrls.profile;
  constructor(private http: HttpClient) {}
  getAll(userId: number): Observable<ApiResponse<SavedSearchResponse[]>> {
    return this.http.get<ApiResponse<SavedSearchResponse[]>>(`${this.baseUrl}/profiles/${userId}/saved-searches`);
  }
  create(userId: number, req: SavedSearchRequest): Observable<ApiResponse<SavedSearchResponse>> {
    return this.http.post<ApiResponse<SavedSearchResponse>>(`${this.baseUrl}/profiles/${userId}/saved-searches`, req);
  }
  update(userId: number, id: number, req: SavedSearchRequest): Observable<ApiResponse<SavedSearchResponse>> {
    return this.http.put<ApiResponse<SavedSearchResponse>>(`${this.baseUrl}/profiles/${userId}/saved-searches/${id}`, req);
  }
  delete(userId: number, id: number): Observable<ApiResponse<void>> {
    return this.http.delete<ApiResponse<void>>(`${this.baseUrl}/profiles/${userId}/saved-searches/${id}`);
  }
}

File: user-profile-service/src/main/java/com/smarttender/profile/service/SavedSearchService.java
package com.smarttender.profile.service;
import com.smarttender.profile.dto.request.SavedSearchRequest;
import com.smarttender.profile.dto.response.SavedSearchResponse;
import java.util.List;
public interface SavedSearchService {
    SavedSearchResponse create(Long userId, SavedSearchRequest request);
    List<SavedSearchResponse> getByUser(Long userId);
    SavedSearchResponse update(Long id, Long userId, SavedSearchRequest request);
    void delete(Long id, Long userId);
}
